app/src/main/java/com/andrew/tasky/agenda/data/event/EventPhotoCompressor.kt

```diff
@@ -0,0 +1,64 @@
+package com.andrew.tasky.agenda.data.event
+
+import com.andrew.tasky.agenda.data.util.BitmapConverters
+import com.andrew.tasky.agenda.domain.models.EventPhoto
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.coroutines.supervisorScope
+import kotlinx.coroutines.withContext
+import okhttp3.internal.toImmutableList
+
+class EventPhotoCompressor {
+
+    private val bitmapConverter = BitmapConverters
+
+    data class CompressPhotosResult(
+        val photos: List<EventPhoto>,
+        val deletedPhotosCount: Int,
+        val deletedPhoto: List<EventPhoto.Local>
+    )
+
+    suspend fun compressPhotos(
+        photos: List<EventPhoto>,
+        compressionSize: Int
+    ): CompressPhotosResult {
+        return withContext(Dispatchers.Default) {
+            var deletedPhotosCount = 0
+            val deletedPhotos = emptyList<EventPhoto.Local>().toMutableList()
```

Comment:
There's already a function to create a mutable list: `val deletedPhotos = mutableListOf<EventPhoto.Local>()`